File: 035_Quiz - scanner/src/hr/unipu/java/Main.java
package hr.unipu.java;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        // Kreiranje objekta 'question', zapis reference u varijablu
        ShortAnswerQuestion question = new ShortAnswerQuestion("U kojem jeziku se programira za Android?", "Java");

        // Poziv medota nad objektom 'question'. Ispis pitanja.
        System.out.println(question.getText());

        // Učitavanje odgovora s tipkovnice
        Scanner scanner = new Scanner(System.in);

        boolean isCorrectAnswer = false;

        // Ponavaljanje učitavanja odgovora s tipkovnice dok odgovor nije točan
        while(!isCorrectAnswer){
            String answer = scanner.nextLine();
            isCorrectAnswer = question.isCorrectAnswer(answer);
            if (isCorrectAnswer) {
                System.out.println("Točno!!!");
            } else {
                System.out.println("Netočno.");
            }
        }

        // Predaja odgovora
//        System.out.println("C: " + question.isCorrectAnswer("C"));
//        System.out.println("Java: " + question.isCorrectAnswer("Java"));
    }
}

File: 035_Quiz - scanner/src/hr/unipu/kotlin/Main.kt
package hr.unipu.kotlin

import java.util.Scanner

fun main(args: Array<String>) {
    // Kreiranje objekta 'question', zapis reference u varijablu
    val question = ShortAnswerQuestion("U kojem jeziku se programira za Android?", "Java")

    // Poziv medota nad objektom 'question'. Ispis pitanja.
    println(question.text)

    // Učitavanje odgovora s tipkovnice
    val scanner = Scanner(System.`in`)

    var isCorrectAnswer = false

    // Ponavaljanje učitavanja odgovora s tipkovnice dok odgovor nije točan
    while (!isCorrectAnswer) {
        val answer = scanner.nextLine()
        isCorrectAnswer = question.isCorrectAnswer(answer)
        if (isCorrectAnswer) {
            println("Točno!!!")
        } else {
            println("Netočno.")
        }
    }

}
